assembl/graphql/utils.py

```diff
@@ -189,3 +192,46 @@ def update_attachment(discussion, attachment_model, new_value, attachments, atta
             attachmentPurpose=attachment_purpose
         )
         attachments.append(attachment)
+
+
+def get_posts_for_phases(discussion, identifiers):
+    """Return related posts for the given phases `identifiers` on `discussion`.
+    """
+    # Retrieve the phases with posts
+    identifiers_with_posts = [i for i in identifiers if i in PHASES_WITH_POSTS]
+    if not discussion or not identifiers_with_posts:
+        return None
+
+    ideas = []
+    # If survey phase, we need the root thematic
+    if Phases.survey.value in identifiers_with_posts:
+        root_thematic = get_root_thematic_for_phase(discussion, Phases.survey.value)
+        if root_thematic:
+            ideas.append(root_thematic)
+
+        identifiers_with_posts.remove(Phases.survey.value)
+
+    if identifiers_with_posts:
+        is_multi_columns = Phases.multiColumns.value in identifiers_with_posts and \
+            len(identifiers_with_posts) == 1
+        ideas.extend(
+            get_ideas(
+                discussion.id,
+                Phases.multiColumns.value if is_multi_columns else None
+            ).all()
+        )
```

Comment:
If you call `get_posts_for_phases(discussion, ['thread', 'multiColumns']` you will have duplicate ideas because ideas from the multiColumns phase are actually all the ideas of the thread phase that have multi columns enabled. You should use a set, not a list to remove duplicate ideas here.

Reply:
Please see get_ideas.